include/swift/SIL/SwiftObjectHeader.h

```diff
@@ -0,0 +1,32 @@
+//===--- SwiftObjectHeader.h - Defines SwiftObjectHeader ------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+#ifndef SWIFT_SIL_SWIFTOBJECTHEADER_H
+#define SWIFT_SIL_SWIFTOBJECTHEADER_H
+
+#include "BridgedSwiftObject.h"
+
+/// The C++ version of SwiftObject.
+///
+/// In C++ code, never use BridgedSwiftObject directly. SwiftObjectHeader has
```

Comment:
I really would like somewhere a large description of how the bridging works that is centralized in the code base. One of those long in source comments. It IMO should describe how we rely on layout compatibility for correctness and also all of the little important bits like this. That way the whole model/its interesting parts is made explicit in the source.

Reply:
It's explained in libswift/README.md. I'll mention it here.